driver-core/src/main/com/mongodb/operation/CommandOperationHelper.java

```diff
@@ -895,6 +895,10 @@ static boolean isRetryableException(final Throwable t) {
             return true;
         }
         String errorMessage = t.getMessage();
+        if (((MongoException) t).getCode() == 20 && errorMessage.contains("Transaction numbers")) {
+            throw new MongoException(((MongoException) t).getCode(), "This MongoDB deployment does not support retryable writes. "
```

Comment:
Change to:

    throw new MongoClientException("This....", t)

Reply:
Done.